src/inference/include/openvino/runtime/intel_gna/gna_properties.hpp

```diff
@@ -0,0 +1,213 @@
+        return os << "GNA_AUTO";
+    case ExecutionMode::HW:
+        return os << "GNA_HW";
+    case ExecutionMode::HW_WITH_SW_FBACK:
+        return os << "GNA_HW_WITH_SW_FBACK";
+    case ExecutionMode::SW_EXACT:
+        return os << "GNA_SW_EXACT";
+    case ExecutionMode::SW_FP32:
+        return os << "GNA_SW_FP32";
+    default:
+        throw ov::Exception{"Unsupported execution mode!"};
+    }
+}
+
+inline std::istream& operator>>(std::istream& is, ExecutionMode& execution_mode) {
+    std::string str;
+    is >> str;
+    if (str == "GNA_AUTO") {
+        execution_mode = ExecutionMode::AUTO;
+    } else if (str == "GNA_HW") {
+        execution_mode = ExecutionMode::HW;
+    } else if (str == "GNA_HW_WITH_SW_FBACK") {
+        execution_mode = ExecutionMode::HW_WITH_SW_FBACK;
+    } else if (str == "GNA_SW_EXACT") {
+        execution_mode = ExecutionMode::SW_EXACT;
+    } else if (str == "GNA_SW_FP32") {
+        execution_mode = ExecutionMode::SW_FP32;
+    } else {
+        throw ov::Exception{"Unsupported execution mode: " + str};
+    }
+    return is;
+}
+/** @endcond */
+
+/**
+ * @brief Enum to define compile and execution targets
+ */
+enum class TargetDevice {
+    UNKNOWN = 0,
+    GNA_2_0 = 1,
+    GNA_3_0 = 2,
+};
+
+/** @cond INTERNAL */
+inline std::ostream& operator<<(std::ostream& os, const TargetDevice& target_device) {
+    switch (target_device) {
+    case TargetDevice::UNKNOWN:
+        return os << "";
+    case TargetDevice::GNA_2_0:
+        return os << "GNA_TARGET_2_0";
+    case TargetDevice::GNA_3_0:
+        return os << "GNA_TARGET_3_0";
+    default:
+        throw ov::Exception{"Unsupported target device!"};
+    }
+}
+
+inline std::istream& operator>>(std::istream& is, TargetDevice& target_device) {
+    std::string str;
+    is >> str;
+    if (str.empty()) {
+        target_device = TargetDevice::UNKNOWN;
+    } else if (str == "GNA_TARGET_2_0") {
+        target_device = TargetDevice::GNA_2_0;
+    } else if (str == "GNA_TARGET_3_0") {
+        target_device = TargetDevice::GNA_3_0;
+    } else {
+        throw ov::Exception{"Unsupported target device: " + str};
+    }
+    return is;
+}
+/** @endcond */
+
+/**
+ * @brief GNA proc_type setting that should be one of AUTO, HW, GNA_HW_WITH_SW_FBACK,
+ *        GNA_SW_EXACT or SW_FP32
+ */
+static constexpr Property<ExecutionMode> execution_mode{"GNA_DEVICE_MODE"};
+
+/**
+ * @brief The option to override the GNA HW execution target. May be one of TARGET_2_0, TARGET_3_0.
+ * By default (in case of no value set) the behavior depends on GNA HW availability:
+ * If GNA HW is present, use the option corresponding to this HW.
+ * If HW is not present, use the option corresponding to the latest fully supported GNA HW generation.
+ * A fully supported GNA HW generation means it must be supported by both the OV GNA Plugin and the core GNA Library.
+ * For the OV GNA Plugin 2022.1, the latest supported GNA HW generation corresponds to GNA_TARGET_3_0.
```

Comment:
I think we should remove the exact version number - it's already outdated. Let's just say "Currently, the latest supported..."